projects/js-packages/svelte-data-sync-client/src/DataSync.ts

```
@@ -0,0 +1,286 @@
+	 * @param valueSchema - The Zod schema to validate the value against.
+	 * @returns The parsed value.
+	 */
+	private getWindowValue< V extends z.ZodSchema >(
+		valueName: string,
+		valueSchema: V
+	): ParsedValue< V > {
+		const validator = z.object( {
+			value: valueSchema,
+			nonce: z.string(),
+		} );
+
+		/**
+		 * This function handles fetching nonces if they're there
+		 * But it shouldn't trigger errors if the nonce is missing.
+		 * Allowing the client application to decide
+		 * how to handle missing nonces.
+		 *
+		 * This is useful, for example, if the `valueSchema` is set to something
+		 * like `z.string().optional()`. In this case, the application might be
+		 * okay with a missing value, so a missing nonce would trigger an unexpected
+		 * error.
+		 */
+		const source = window[ this.namespace ][ valueName ]
+			? window[ this.namespace ][ valueName ]
+			: { value: undefined, nonce: '' };
+
+		const result = validator.parse( source );
+		return result as ParsedValue< V >;
+	}
+
+	private async request(
+		method: RequestMethods,
+		partialPathname: string,
+		params?: RequestParams,
+		abortSignal?: AbortSignal
+	) {
+		const url = `${ this.wpDatasyncUrl }/${ partialPathname }`;
+		const args: RequestInit = {
+			method,
+			signal: abortSignal,
+			headers: {
+				'Content-Type': 'application/json',
+				'X-WP-Nonce': this.wpRestNonce,
+				'X-Jetpack-WP-JS-Sync-Nonce': this.endpointNonce,
+			},
+			credentials: 'same-origin',
+			body: null,
+		};
+
+		if ( method === 'POST' ) {
+			args.body = JSON.stringify( { JSON: params } );
+		}
+
+		const result = await this.attemptRequest( url, args );
+
+		let data;
+		const text = await result.text();
+		try {
+			data = JSON.parse( text );
+		} catch ( e ) {
+			// eslint-disable-next-line no-console
+			console.error( 'Failed to parse the response\n', { url, text, result, error: e } );
+			throw new ApiError( url, 'json_parse_error', 'Failed to parse the response' );
```

While going up the chain where request is used, I noticed there's no try catch for errors like this one.

---

When a request fails - it should throw an error, whoever is triggering the method should catch it.

Most of the time it's going to be caught by SyncedStore.ts to retry the request.

---

Ah okay - thank you! Missed that 😅